### DISCUSSION TITLE
Changes to the partitioning of stem wood to CWD classes

### DISCUSSION THREAD

**Discussion by adamhb** [NGEET/fates#962] [Ideas]
In recent CA FATES meetings we've talked about the fuel dynamics for simulations with grasses and shrubs. Charlie and Zachary have implemented some recent changes for grasses. For woody pfts, my understanding of the main branch of fates is that it [sends a fixed fraction](https://github.com/NGEET/fates/blob/2f90901cb7f1758766111442e6d40eefcdc564f9/biogeochem/EDPhysiologyMod.F90#L2438) of structural wood (aka deadwood) and sapwood to the "trunk" fuel class (i.e. 1000 hr fuel class) based on the SF_val_CWD_frac parameter (aka "fates_frag_cwd_frac" in the fates param file). I'm running with shrubs, and the code currently treats structural/stem wood from shrubs as 1000-hr fuels. Therefore, when a small shrub (or small tree) experiences turnover or dies, part of its trunk becomes 1,000 hr fuel and doesn't burn. This makes sense for large logs, but perhaps not for shrub stems or small trees...

I updated the code in one of my branches such that cohorts below a size threshold (for now I made this a pft parameter) don't send any stem wood to trunk fuel (it goes to large branch fuel instead, and therefore will burn). I made this change to all subroutines that I'm aware of where CWD is divided into these fuel classes: CWDInput (ED physiology), mortality_litter_fluxes (patch dynamics), fire_litter_fluxes (patch dynamics), logging_litter_fluxes (logging), and SendCohortToLitter (cohort dynamics).

I'm curious if A) I've missed any other places, aside from the subroutines above, where CWD is divided into these fuel classes? and B) if anyone has thoughts / feedback on this change in general?

After the changes I've made so far you can see that the accumulation of trunk fuel happens more slowly (as expected).
![results_trunk_fuel_fix](https://user-images.githubusercontent.com/25392545/208180076-a7f5f316-6622-4f8e-bb16-7c7c9b99ab7a.png)

**Comment by rosiealice** [comment]
This makes sense to me, esp given the difference in how trunk fuel is treated.

**Reply by rosiealice** [reply]
...also though, this will affect the dynamics of calculating weighted average properties of the fuel, for the cases where trunks are excluded from the calculations (bulk density, SAB etc. Line. 572 or thereabouts in SfMain) 

More lb would be created and more would thus go into that calculation, and (because that class is less flammable) the overall flammability of the fuel will likely go down? 

Just thinking this through. It might be interesting to plot those things out to see what happens?

**Comment by ckoven** [comment]
Your plan @adamhb makes sense to me too. I guess a more sophisticated version could have a continuous function of fuel size class fractions as dependent on DBH.  It does seem that with the rapid expansion of terrestrial lidar data, that this could be measured and parameterized in a pretty direct way for any given ecosystem. But the simple version already seems like it would make more sense than what is currently there!

**Comment by jkshuman** [comment]
@adamhb it is a good improvement, and makes sense. I haven't looked at all the routines, but can do that on the PR when it comes across. The fuel size classes come from Rothermal's fire spread work (Rothermal et al 1983). It was based off of work by Fosberg et al. (1971) for timelag in fuel moisture calculations for fire danger rating. The Fosberg (1971) work compared well for ponderosa pine fuel sticks, and is the fuel size class breakdown which is still used today by forest service. My recommendation would be to use those classes as an initial justification for identifying the switch from trunk fuel to large branch fuel. So reducing that DBH value to 7.6 cm. 

twigs (1-hour fuels) = <0.6 cm diameter
small branches (10-hour fuels) = 0.6 to 2.5 cm diameter
large branches (100-hour) = 2.5 to 7.6 cm diameter
trunks (1000-hour fuels) = more than 7.6 cm diameter

**Comment by adamhb** [comment]
Thanks for the great thoughts @rosiealice, @ckoven, and @jkshuman. Glad to hear this makes sense to others. Rosie, good thought re. the weighted average properties of the fuel-- I'll plot how those variables change with this modification. Jackie, thanks for the fuel class diameter breakdown and refs-- super helpful! Will move the threshold down to 7.6 cm (10 was just an initial value and it did seem a little too high to me).

**Reply by jkshuman** [reply]
@adamhb worth mentioning that these fuel classes also map to the parameter for maxdecomp. So, when adjusting the maxdecomp parameter values you can use the same diameter size classes. I found in the literature decomp for things below 10cm were classified as fine woody debris (FWD), and then there were rates for stem decomp. The default param value for maxdecomp is for work in the tropics.

I added the names to the fuel classes above (and here again)
twigs (1h) - <0.6 cm diameter 
small branch (10h) - 0.6 to 2.5 cm diameter 
large branch (100h) - 2.5 to 7.6 cm diameter 
stem (1000h) - greater than 7.6 diameter 
dead leaves 
live grass - (no decomp as these are alive)

**Reply by adamhb** [reply]
@jkshuman Thanks a lot for bringing this up. I've been using the default so far for CA simulations, but should change that. It looks like the max decomposition rate param will affect the realized decomposition rates quite a lot in the model given how the algorithm is described in the tech doc (the max rate is used as the starting point which is then "ameliorated" based on temp and moisture).

@ckoven, your max decomp rates for CA simulations seem to be higher than the (tropical) default. Do I have that right?

**Comment by adamhb** [comment]
Hi all,

I updated the code on this development branch to incorporate feedback from you all (thanks @jkshuman @rosiealice @ckoven). Stem wood is now partitioned to CWD pools based on cohort dbh and the Rothermel fuel size class boundaries. For example, if a cohort with dbh of 5 cm dies, then the fraction of its fuel that previously would have gone to 1,000 hr fuel goes to 100 hr instead. If a 2 cm dbh cohort dies then the fractions of its stem wood that would have gone to 1000hr and 100hr go to 10 hr fuel instead.

The [results](https://docs.google.com/presentation/d/1_2a65PYSeXfy97NKQy06QQiF19EP5TSKgJr_eCPRRk0/edit?usp=sharing) are pretty much as expected. There is less build up of 1,000 hr fuels in young forests and more going to the smaller fuel classes. These differences are most obvious in the early forest development stages (first 20 years of a bare ground spin up).

As @rosiealice hypothesized the fuel is slightly less flammable (lower SAV and bulkd) due to more weighting of the small and large branch fuel classes compared to the other flammable fuel classes.

Let me know if you all have any additional thoughts or feedback!
best,
Adam

**Reply by jkshuman** [reply]
that's a great update @adamhb Thanks for sharing the details.

**Reply by rosiealice** [reply]
Awesome! If you happen to make any figures on the impact then those would be cool to store somewhere (if not in a paper then in the PR maybe?)

**Comment by glemieux** [comment]
I'm temporarily locking this thread as a github member not affiliated with our organization is getting notifications from a now blocked user that was spamming this thread, apparently weeks after the fact.
